### PR TITLE
update e2e test + cleanup

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Update whitelist e2e test to ensure mixed scenario works as expected
- Add unit tests
- Cleanup
- Add comments 

**Issue Fixed**:

**Notes for Reviewers**:
Addressing comments from previous PR

### CODE REVIEW

**Comment by aramase** [comment]
```bash
Ran 15 of 15 Specs in 2664.251 seconds
SUCCESS! -- 15 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAADPodIdentity (2664.25s)
PASS
ok  	github.com/Azure/aad-pod-identity/test/e2e	2664.280s
```

File: test/e2e/aadpodidentity_test.go [hunk 1]
```diff
@@ -708,42 +708,94 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
...
+
 		// Assign user assigned identity to every node
 		nodeList, err := node.GetAll()
 		Expect(err).NotTo(HaveOccurred())
 
-		enableUserAssignedIdentityOnCluster(nodeList, keyvaultIdentity)
+		enableUserAssignedIdentityOnCluster(nodeList, fmt.Sprintf("%s-%d", keyvaultIdentity, 1))
+		enableUserAssignedIdentityOnCluster(nodeList, fmt.Sprintf("%s-%d", keyvaultIdentity, 2))
+
+		enableSystemAssignedIdentityOnCluster(nodeList)
```
Comment (kkmsft) [inline test/e2e/aadpodidentity_test.go]: Not sure why we are enabling system assigned identity in this test ?

Comment (kkmsft) [reply test/e2e/aadpodidentity_test.go]: How do we get this to work with a setup with system assigned identity cluster (assuming that #319 is through) ?

File: test/e2e/identityvalidator/identityvalidator.go [hunk 2]
```diff
@@ -87,8 +87,12 @@ func testClusterWideUserAssignedIdentity(logger *log.Entry, msiEndpoint, subscri
 
 // testUserAssignedIdentityOnPod will verify whether a pod identity is working properly
 func testUserAssignedIdentityOnPod(logger *log.Entry, msiEndpoint, identityClientID, keyvaultName, keyvaultSecretName, keyvaultSecretVersion string) error {
+	// When new authorizer is created, it tries to create dataplane authorizer using MSI. It checks the AZURE_CLIENT_ID to get the client id
```
Comment (kkmsft) [inline test/e2e/identityvalidator/identityvalidator.go]: Can you please specifiy about the auth library instead of 'it'.

**Pull request merged**